### PR TITLE
ADD (Core) Wp table add in orientation and delay collumn TC Cherrypick

### PR DESCRIPTION
**Pull request by acidmanifesto**
ADD (Core) Wp table add in orientation and delay collumn
TC Cherry pick of https://github.com/TrinityCore/TrinityCore/commit/4b7d19c91bde2cabb8498809dee935c7efb2660a

## Changes Proposed:
-  adds in the orientation and delay to the waypoints table in the world.
-  Request made of Malcrom
 
## SOURCE:
https://github.com/TrinityCore/TrinityCore/commit/4b7d19c91bde2cabb8498809dee935c7efb2660a

## Tests Performed:
- Builds and Compiles with no issues

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by acidmanifesto** [comment]
Need review

File: src/server/game/AI/SmartScripts/SmartScriptMgr.h [hunk 1]
```diff
@@ -28,18 +28,22 @@
 
 struct WayPoint
 {
-    WayPoint(uint32 _id, float _x, float _y, float _z)
+    WayPoint(uint32 _id, float _x, float _y, float _z, float _o, uint32 _delay)
```
Comment (Nefertumm) [inline src/server/game/AI/SmartScripts/SmartScriptMgr.h:31]: Is this struct used somewhere else? If yes, it should be better to set the last parameters to 0 as default.

**Review by Nefertumm — APPROVED** [review]
All good!

File: data/sql/updates/db_world/rev_1635951187866923861.sql [hunk 2]
```diff
@@ -0,0 +1,4 @@
+-- add revision
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1635951187866923861');
+-- add orientation` and delay into existing wp table
+ALTER TABLE `waypoints` ADD COLUMN `orientation` FLOAT DEFAULT 0 NOT NULL AFTER `position_z`, ADD COLUMN `delay` INT(10) UNSIGNED DEFAULT 0 NOT NULL AFTER `orientation`;
```
[the remark this answers is not in the document]
Comment (acidmanifesto) [reply data/sql/updates/db_world/rev_1635951187866923861.sql:4]: I am going to be the first person to tell ya, i am not a sql person at all, and a very weak at it. at this point i did and ported over, and in this case copy-pasta the sql from the referenced tc pr https://github.com/TrinityCore/TrinityCore/commit/4b7d19c91bde2cabb8498809dee935c7efb2660a

File: src/server/game/AI/SmartScripts/SmartScriptMgr.h (hunk 1 above)
Comment (acidmanifesto) [reply src/server/game/AI/SmartScripts/SmartScriptMgr.h:31]: this struct was throwinging up the compile test errors, apparrently it is related to the adjustments in adding the orentation and the delay which is definately a bit different from what hte original pr on tc showed. at this point, it is required for compile testing from what i can tell. I do not use linux or macos, so the error present on those i would not be able to verify first hand.

File: data/sql/updates/db_world/rev_1635951187866923861.sql (hunk 2 above)
[the remark this answers is not in the document]
Comment (Kitzunu) [reply data/sql/updates/db_world/rev_1635951187866923861.sql:4]: Just use INT, skip the width. It is deprecsted in later versions of MySql 8.0

**Review by Kitzunu — CHANGES_REQUESTED** [review]

File: src/server/game/AI/SmartScripts/SmartScriptMgr.h [hunk 3]
```diff
@@ -28,18 +28,22 @@
...
+        o = _o;
+        delay = _delay;
     }
 
     uint32 id;
     float x;
     float y;
     float z;
+    float  o;
+    uint32 delay;
```
[the remark this answers is not in the document]
Comment (Kitzunu) [reply src/server/game/AI/SmartScripts/SmartScriptMgr.h:46]: It's probably due to that dumb clang formatter lol

File: data/sql/updates/db_world/rev_1635951187866923861.sql (hunk 2 above)
[the remark this answers is not in the document]
Comment (acidmanifesto) [reply data/sql/updates/db_world/rev_1635951187866923861.sql]: done

**Comment by Nyeriah** [comment]
SQL files need to go to the *pending* folder, the importer scripts takes care of moving them to the *world* folder. In a lapse, I didn't check if the folder was correct before merging this.

**Pull request merged**